### PR TITLE
Inject breakpoint-markers into github

### PR DESCRIPTION
**Pull request by jamohile**
**Background**
For MVP, the only way to create a breakpoint was to manually enter a file name and line number. This is (for obvious reasons) clunky, and was never the final planned implementation.

**Work Done**
Injected markers into github to set/view breakpoints.
image
This image shows the following state:
1. create-new, not hovered.
2. create-new, hovered.
3. active (pending)
4. completed

**Future Work**
- Improved integration of breakpoint results, e.g more data on marker-click in a popup
- More test coverage, will require mocking DOM querySelector
- Adding support for non-github sites.

**NOTE: This commit also shows changes from non-feature related lines, particularly in injected-app.tsx. This is due to the prettier formatting**

### CODE REVIEW

File: app/src/client/injected/injected-app.tsx [hunk 1]
```diff
@@ -177,10 +178,32 @@ interface InjectedAppState{
...
         <Input placeholder="Line Number " />
         <Button primary onClick={this.createBreakPoint(this.state.lineNumber)}> CREATE </Button> */}
+        <BreakpointMarkers createBreakpoint={(fileName, lineNumber) => this.createBreakPoint(fileName, lineNumber)}/>
       </>
     );
   }
+}
+
+
+const BreakpointMarkers = ({createBreakpoint}) => {
```
Comment (KushalD1999) [inline app/src/client/injected/injected-app.tsx]: Missing Documentation

File: app/src/client/markers/BreakpointMarker.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import styled from "styled-components";
+
+import { BreakpointMeta, Breakpoint } from "../../common/types/debugger";
+
+interface NewBreakpointMarkerProps {
```
Comment (KushalD1999) [inline app/src/client/markers/BreakpointMarker.tsx:6]: I think documentation for all this interfaces would be really helpful

File: app/src/client/injected/injected-app.tsx (hunk 1 above)
Comment (jamohile) [reply app/src/client/injected/injected-app.tsx]: Refactored and documented this.

File: app/src/client/markers/BreakpointMarker.tsx (hunk 2 above)
Comment (jamohile) [reply app/src/client/markers/BreakpointMarker.tsx:6]: Added

File: app/__tests__/markers/breakpointMarkers.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+configure({ adapter: new Adapter() });
+
+describe("NewBreakpointMarker", () => {
+  it("responds to clicks", () => {
+    const spy = jest.fn();
+    const wrapper = shallow(<NewBreakpointMarker onClick={spy}/>);
+    wrapper.simulate("click");
+    expect(spy).toHaveBeenCalled();
+  });
+});
```
[the remark this answers is not in the document]
Comment (jamohile) [reply app/__tests__/markers/breakpointMarkers.tsx:15]: Added!

File: app/src/client/injected/injected-app.tsx [hunk 4]
```diff
@@ -1,116 +1,121 @@
...
-    /**
-   * This function is invoked immediately after a component is mounted. This allows to 
-   * keep calling listbreakpoints request on an interval of 5 seconds to check 
+  /**
+   * This function is invoked immediately after a component is mounted. This allows to
+   * keep calling listbreakpoints request on an interval of 5 seconds to check
    * the difference between active breakpoints and non-active breakpoints.
    */
-  componentDidMount(){
+  componentDidMount() {
```
[the remark this answers is not in the document]
Comment (jamohile) [reply app/src/client/injected/injected-app.tsx:118]: Yep!

**Pull request merged**